### PR TITLE
Unexpected ignored self key request when it's not shared history

### PR DESCRIPTION
**Pull request by mcalinghee**
### Checklist
- [ ] Tests written for new code (and old code if feasible)
- [ ] Linter and other CI checks pass
- [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

Signed-off-by: maghen.calinghee@beta.gouv.fr

### Step to reproduce the issue
**Scenario 1: Verification of 2 sessions when logging : session A in and session B in**
_Contexte:_
matrix-js-sdk : 20.0.0
matrix-react-sdk : 3.57.0

1. user 1 has already a session opened : session A
2. user 1 is opening a new session by logging : session B
3. on session B, a verification request dialog is opening
4. on session A, a verification dialog is waiting 
5. session B press `Continue`
6. session A displayed the verification process dialog
7. session B displayed the verification process dialog
8. session A press `They match` and waits for session B
9. session B press `They match`, a success dialog is displayed 
10. on session A , a success dialog is displayed 
11. session A press `Got it`
12. session B receive the key and decrypt the messages in a **direct message**
Screenshot 2022-10-03 at 10 38 13

13. session B **cannot** decrypt the messages in a **room**
Screenshot 2022-10-03 at 10 38 57

14. **Re-requesting keys is not working as well**

 => Test Result : KO 🔴 

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Unexpected ignored self key request when it's not shared history ([\#2724](https://github.com/matrix-org/matrix-js-sdk/pull/2724)). Contributed by @mcalinghee.

### CODE REVIEW

**Comment by mcalinghee** [comment]
After a discussion with @giomfo, we have seen that we might have a bug when processing the forwarded key request(`m.forwarded_room_key`), it is possible for someone to have a look at this?

**Review by duxovni — APPROVED** [review]

**Comment by weeman1337** [comment]
@mcalinghee thanks for you contribution. Can you check the [linter report](https://github.com/matrix-org/matrix-js-sdk/actions/runs/3180636037/jobs/5186641988) and fix these issues? Also please merge `develop` into the branch.

**Review by weeman1337 — APPROVED** [review]

File: src/crypto/algorithms/megolm.ts [hunk 1]
```diff
@@ -1455,7 +1456,7 @@ class MegolmDecryption extends DecryptionAlgorithm {
                 // If someone sends us an unsolicited key for a room
                 // we're already in, and they're not one of our other
                 // devices or the one who invited us, ignore it
-                if (room && !fromInviter && !fromUs) {
+                if (room && !fromInviter) {
                     logger.log("forwarded key not from inviter or from us - ignoring");
```
Comment (richvdh) [inline src/crypto/algorithms/megolm.ts:1460]: > or from us

and nor is this log line

**Review by richvdh — COMMENTED** [review]
It's disappointing that no care was taken to update the comments or log lines to match the new behaviour of the code. Poor quality changes like this make maintenance of the software harder and this should have been caught in review.

This is *particularly* important in security-related code.

**Comment by richvdh** [comment]
ohhh. I think I owe everyone an apology. I've just re-read the code properly and seen the `!fromUs` condition is just moved higher up, so those lines are still valid.

I'm really sorry. I'll get back under my bridge.

**Pull request merged**